lib/galaxy/security/validate_user_input.py

```diff
@@ -33,6 +33,8 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False):
         return message
     if not(VALID_EMAIL_RE.match(email)):
         message = "The format of the email address is not correct."
+    elif not email.islower():
+        message = "Please use lower-case characters for your email."
```

Comment:
This function is used in several places and I don't think we can just assume that Galaxy databases don't contain mixed-case email addresses. I have an alternative approach in the works though, should be finished soon.

Reply:
This function will only prevent that new upper-case emails are entered into the database. It should not effect existing upper-case emails already stored. I think it is the correct solution for this problem.